src/ui/components/IconStar/index.js

```diff
@@ -0,0 +1,131 @@
+    case OPEN_STYLE:
+    default:
+      return 'M317.994575,670.99995 C316.704598,671.000763 315.477615,670.442079 314.630967,669.468394 C313.784319,668.49471 313.401158,667.201652 313.580582,665.923653 L316.046749,648.259919 L304.193762,635.514481 C303.080773,634.318044 302.711733,632.608076 303.232152,631.058811 C303.752571,629.509546 305.078939,628.369589 306.688275,628.088421 L323.214424,625.130961 L331.013827,609.468577 C331.767364,607.955994 333.3113,607 335.000594,607 C336.689888,607 338.233824,607.955994 338.98736,609.468577 L346.790813,625.130961 L363.329111,628.08437 C364.934946,628.371492 366.25546,629.513805 366.771316,631.062053 C367.287172,632.610301 366.915846,634.316807 365.803377,635.51043 L353.954439,648.26397 L356.420606,665.923653 C356.652457,667.578241 355.93975,669.223573 354.574418,670.185702 C353.209085,671.147831 351.420524,671.265104 349.941351,670.489485 L335.002619,662.698806 L320.047688,670.50569 C319.413201,670.833752 318.708782,671.003331 317.994575,670.99995 Z M314.678006,634.89463 L324.603415,645.569846 L322.578647,660.041143 L335.002619,653.56309 L347.42254,660.045194 L345.397773,645.573897 L355.323182,634.89463 L341.352288,632.39902 L335.002619,619.637378 L328.648899,632.39902 L314.678006,634.89463 Z';
+  }
+};
+
+export type Props = {|
+  alt?: $PropertyType<IconProps, 'alt'>,
+  className?: string,
+  half?: boolean,
+  selected?: boolean,
+  readOnly?: boolean,
+  yellow?: boolean,
+|};
+
+const IconStar = ({
+  className,
+  half = false,
+  selected = false,
+  readOnly = false,
+  yellow = true,
+  ...iconProps
+}: Props) => {
+  let color = YELLOW;
+
+  if (!yellow) {
+    color = GRAY;
+  }
+
+  let starStyle = selected ? CLOSED_STYLE : OPEN_STYLE;
+
+  if (half) {
+    starStyle = HALF_STYLE;
+  }
+
+  if (readOnly && !selected) {
+    starStyle = DIM_CLOSED_STYLE;
+  }
+
+  let defs;
+  let gProps = {
+    fill: color,
+    fillRule: 'nonzero',
+  };
+
+  switch (starStyle) {
+    case CLOSED_STYLE:
+    case DIM_CLOSED_STYLE:
+      gProps = {
+        ...gProps,
+        transform: 'translate(-140.000000, -607.000000)',
+        fillOpacity: starStyle === DIM_CLOSED_STYLE ? 0.25 : 1,
+      };
+      break;
+    case HALF_STYLE: {
+      // Key is needed in case there are multiple IconStars on 1 page.
+      const key = Math.random();
```

Comment:
This was 1 of the reasons why the half star didn't show up more than once on the home page (and other place where there could be multiple stars), because this is used as part of the id which needs to be unique.